api/src/main/java/com/messagebird/objects/conversations/MessageComponentType.java

```diff
@@ -10,8 +13,18 @@ public enum  MessageComponentType {
   FOOTER("footer"),
   BUTTON("button"),
   CARD("card"),
-  CAROUSEL("carousel");
+  CAROUSEL("carousel"),
+  LIMITED_TIME_OFFER("limited_time_offer");
 
+  private static final Map<String, MessageComponentType> TYPE_MAP;
+
+  static {
+    Map<String, MessageComponentType> map = new HashMap<>();
```

Comment:
Since you are mapping from MessageComponentType (an enum) to String, it is often efficient to use EnumMap instead of HashMap

Reply:
I opted to use a HashMap instead of an EnumMap because we are mapping from a String (the lowercase string representation of the enum value) to the MessageComponentType enum. Since EnumMap is specifically designed for enum-to-something mappings, it wouldn’t fully apply in this case where our key is a String. Using a HashMap allows us to efficiently map the string values to the corresponding enum, providing O(1) for the lookup performance.